File: SDE_sheet/Arrays/02_pascal_triangle/solution.go
func generate(numRows int) [][]int {
    if numRows == 0 {
        return [][]int{}
    }
    
    nums := make([][]int, numRows)
    nums[0] = []int{1}
    
    for i := 1; i < numRows; i++ {
        nums[i] = make([]int, i+1)
        nums[i][0], nums[i][i] = 1, 1
        
        for j := 1; j < i; j++ {
            nums[i][j] = nums[i-1][j-1] + nums[i-1][j]
        }
    }
    
    return nums
}

File: SDE_sheet/Arrays/02_pascal_triangle/solution.rs
impl Solution {
    pub fn generate(num_rows: i32) -> Vec<Vec<i32>> {
        let mut nums: Vec<Vec<i32>> = Vec::new();
        
        if num_rows == 0 {
            return nums;
        }
        
        nums.push(vec![1]);
        
        for i in 1..num_rows as usize {
            let mut row: Vec<i32> = vec![0; i + 1];
            row[0] = 1;
            row[i] = 1;
            
            for j in 1..i {
                row[j] = nums[i - 1][j - 1] + nums[i - 1][j];
            }
            
            nums.push(row);
        }
        
        nums
    }
}
